tests/unit/amo/components/TestConfirmationDialog.js

```diff
@@ -1,118 +1,85 @@
-    expect(cancelButton).toHaveProp('buttonType', 'cancel');
-    expect(cancelButton.children()).toHaveText('Cancel');
+    const cancelButton = screen.getByRole('button', { name: 'Cancel' });
+    expect(cancelButton).toHaveClass('ConfirmationDialog-cancel-button');
+    expect(cancelButton).toHaveClass('Button--cancel');
   });
 
   it('renders a custom className', () => {
     const className = 'MyComponent';
-    const root = render({ className });
-    expect(root).toHaveClassName(className);
-    expect(root).toHaveClassName('ConfirmationDialog');
+    render({ className });
+    expect(screen.getByClassName('ConfirmationDialog')).toHaveClass(className);
   });
 
-  it.each([true, false])(
-    'can configure buttons as puffy=%s',
-    (puffyButtons) => {
-      const root = render({ puffyButtons });
-
-      expect(root.find('.ConfirmationDialog-cancel-button')).toHaveProp(
-        'puffy',
-        puffyButtons,
-      );
-      expect(root.find('.ConfirmationDialog-confirm-button')).toHaveProp(
-        'puffy',
-        puffyButtons,
-      );
-    },
-  );
-
-  it('calls onConfirm() when user clicks Confirm', () => {
-    const onConfirm = sinon.spy();
-    const root = render({ onConfirm });
+  it('can configure buttons as puffy', () => {
+    render({ puffyButtons: true });
 
-    const event = createFakeEvent();
-    root.find('.ConfirmationDialog-confirm-button').simulate('click', event);
+    const cancelButton = screen.getByRole('button', { name: 'Cancel' });
+    expect(cancelButton).toHaveClass('ConfirmationDialog-cancel-button');
+    expect(cancelButton).toHaveClass('Button--puffy');
 
-    sinon.assert.calledWith(onConfirm, event);
+    const confirmButton = screen.getByRole('button', { name: 'Confirm' });
+    expect(confirmButton).toHaveClass('ConfirmationDialog-confirm-button');
+    expect(confirmButton).toHaveClass('Button--puffy');
   });
 
-  it('calls onCancel() when user clicks Cancel', () => {
-    const onCancel = sinon.spy();
-    const root = render({ onCancel });
+  it('can configure buttons as non puffy', () => {
+    render({ puffyButtons: false });
 
-    const event = createFakeEvent();
-    root.find('.ConfirmationDialog-cancel-button').simulate('click', event);
+    const cancelButton = screen.getByRole('button', { name: 'Cancel' });
+    expect(cancelButton).toHaveClass('ConfirmationDialog-cancel-button');
+    expect(cancelButton).not.toHaveClass('Button--puffy');
 
-    sinon.assert.calledWith(onCancel, event);
+    const confirmButton = screen.getByRole('button', { name: 'Confirm' });
+    expect(confirmButton).toHaveClass('ConfirmationDialog-confirm-button');
+    expect(confirmButton).not.toHaveClass('Button--puffy');
+  });
+
+  it('calls onConfirm() when user clicks Confirm', () => {
+    const onConfirm = jest.fn();
+    render({ onConfirm });
+    const button = screen.getByClassName('ConfirmationDialog-confirm-button');
```

Comment:
Nit/tip: You could just inline the button selector via `userEvent.click(screen.getByClassName('ConfirmationDialog-confirm-button'));`, as we don't do anything else with the button in the test.